### PR TITLE
 Set up the more advanced unhandled exception handler on Application Insights

### PR DESCRIPTION
**Pull request by erbridge**
This should give us more information about exceptions that occur. There is possibly some overlap here with Rollbar.

### CODE REVIEW

**Comment by erbridge** [comment]
[CYBER 4: Logging and alerting](https://trello.com/c/lFxQbjj3/559-cyber-4-logging-and-alerting)

File: config/initializers/application_insights.rb [hunk 1]
```diff
@@ -4,5 +4,7 @@
   Rails.application.configure do
     buffer_size = 1
     config.middleware.use ApplicationInsights::Rack::TrackRequest, instrumentation_key, buffer_size
+
+    ApplicationInsights::UnhandledException.collect(app_insights_key)
```
Comment (pezholio) [inline config/initializers/application_insights.rb]: Does this automatically alert us, or do we need to set up some more alerts?

Comment (erbridge) [reply config/initializers/application_insights.rb]: No, this just populated Application Insights. We will need to set up any alerts in there.

Comment (pezholio) [reply config/initializers/application_insights.rb]: Cool 👍

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Comment by erbridge** [comment]
[Consider using the unhandled exception handler on AI](https://trello.com/c/zVRcgy1w/539-consider-using-the-unhandled-exception-handler-on-ai)

**Pull request merged**